src/vcpkg/vcpkgpaths.cpp

```diff
@@ -134,6 +134,18 @@ namespace vcpkg
         return parsed_config_opt;
     }
 
+    static std::vector<std::string> merge_overlays(const std::vector<std::string>& cli_overlays,
+                                                   const std::vector<std::string>& manifest_overlays,
+                                                   const std::vector<std::string>& env_overlays)
+    {
+        std::vector<std::string> ret = cli_overlays;
+
+        ret.insert(std::end(ret), std::begin(manifest_overlays), std::end(manifest_overlays));
+        ret.insert(std::end(ret), std::begin(env_overlays), std::end(env_overlays));
```

Comment:
Perhaps we could benefit from a `std::vector<T> merge(std::vector<T> v, Args... args)`? 
What does our templates expert @BillyONeal think?